packages/react-components/react-popover/src/components/Popover/Popover.types.ts

```diff
@@ -209,6 +209,7 @@ export type OnOpenChangeData = { open: boolean };
  * The supported events that will trigger open/close of the menu
  */
 export type OpenPopoverEvents =
+  | Event
   | MouseEvent
   | TouchEvent
   | React.FocusEvent<HTMLElement>
```

Comment:
Yeah good catch, I tried playing with this a bit more in TS playground and it can indeed break typescript build depending on how users define their onOpenChange handlers https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBDAnmYcDyKB2AFCkBuwUAooZjAM5wC8AUHHAD5ynDn1NwCyEArhcFbsGzACp8AxgAshMWvKQo42KHirU4Abw4MImDGwDCUgIaYA5sABccABTX0WXASKyKAShoA+OPggBLABMAbloAX3kJPQp4PQNMYzNLGjsHHn5BMngxSRksz2ofbXDI6PgwVTAKGxU1FM04OKxEi1Qw0NogA

Reply:
In this case we might want to consider casting internally in the `Popover` and `Menu` components to one of the existing event types, to unblock the fixing of the bugs in the description. It will change the runtime behaviour, but I think that is acceptable compared to the risk of build breaks for users. 

However actually adding new events to types in `useOnScrollOutside` is also technically a breaking change in that case 😱😱

WDYT @behowell?

Reply:
The original implementation is at this commit: https://github.com/microsoft/fluentui/pull/28964/commits/b73b2bd9310a9bb1ef99b39b28306856dda3d72e

Reply:
Casting to avoid a breaking change in types is still a breaking change; and possibly even worse: it results in extra-confusing runtime bugs when the object isn't the type that was promised. In this case, the event object would be missing any of the properties that are in MouseEvents.

I think in practice, it's pretty unlikely that it will cause an actual problem in this case, especially since the existing type includes `React.FocusEvent<HTMLElement>` and already requires you to do some runtime inspection of the object to see what type it is.

It might be worth discussing this with the rest of the team to come up with the right fix here. This isn't the first time this has caused a problem. In my recent PR https://github.com/microsoft/fluentui/pull/28951, I had to come up with a "hacky" workaround to this problem, by including the new event on the `data` argument instead. Unfortunately, that won't work here because it requires passing `undefined` for the `event` argument, which would be a breaking change here (the `OpenPopoverEvents` type doesn't include undefined as a possibility).

All of the possible options I can think of are broken in some way, so we'd have to decide what the least-bad option is:
1. Add `| Event` to the `OpenPopoverEvents` type.
   - **Pros**: Gets the type correct. Any potential breaks are caught at build time.
   - **Cons**: It's a breaking change for any code that explicitly types its event argument (instead of using our exported `OpenPopoverEvents` type).
2. Cast the event type to `MouseEvent`.
   - **Pros**: No type changes, and no changes to existing code to listen to the new case where `onOpenChange` can be called.
   - **Cons**: Still a breaking change, but the break happens at runtime instead of build time. The type is a lie.
3. Create `onOpenChange2` with the new types, and deprecate `onOpenChange`.
   - **Pros**: No type breaks to existing code.
   - **Cons**: Breaks the contract of the old `onOpenChange`: it won't be called when the popover hides in response to a scrollbar. Code will need to be updated to `onOpenChange2`. Also, this adds a bunch of code to keep both the old event and new event working.

Given all of that, I think option 1 might be the least-bad option, but could ruffle the feathers of users of the library by including a breaking change in a minor version.

Reply:
Yeah I think that 1. might be the least bad option, but it would require discussion with the wider team to make sure we know about it and the consequences. I would propose that we:

1. Implement option 2 (casting the event) to fix the standing bug
2. create an issue to track whether we stick to option 2 or switch to option 1
3. discuss the new issue in tech sync

In the interim option 2 is the best case IMO, since build breaks generally block our users from upgrading to fix other issues. I can't as easily think of cases where users rely on knowing exactly what the event type is. However for option 1 it's quite common from what I see in codebases to type the event

```tsx
👍 no need to do anything
const onOpenChange: PopoverProps['onOpenChange'] = (e, data) => {/**/}

👎 breaking change - I've generally seen this more often since developers generally don't pick from PopoverProps as much
const onOpenChange = (e: React.MouseEvent | React.FocusEvent, data: PopoverOpenData) => {/**/}
```